mpitest/test_parallel_record.py

```diff
@@ -0,0 +1,56 @@
+""" Testing out recorders under MPI."""
+
+import os
+import unittest
+
+from openmdao.core import Problem
+from openmdao.core.mpi_wrap import MPI
+from openmdao.recorders import DumpRecorder
+from openmdao.test.simple_comps import FanInGrouped
+from openmdao.test.mpi_util import MPITestCase
+
+if MPI:
+    from openmdao.core.petsc_impl import PetscImpl as impl
+else:
+    from openmdao.core import BasicImpl as impl
+
+
+class TestDumpRecorder(MPITestCase):
+
+    N_PROCS = 2
+
+    def setUp(self):
+        if MPI:
+            if MPI.COMM_WORLD.rank == 0:
+                self.filename = 'data_0.dmp'
+            elif MPI.COMM_WORLD.rank == 1:
+                self.filename = 'data_1.dmp'
+        else:
+            self.filename = 'data.dmp'
+
+    def tearDown(self):
+        if os.path.exists(self.filename):
+            os.remove(self.filename)
+
+    def test_dump_converge_diverge_par(self):
+
+        prob = Problem(impl=impl)
+        prob.root = FanInGrouped()
+
+        rec = DumpRecorder(out='data.dmp')
```

Comment:
No, it is okay, `data,dmp` is the name the user gives, but under MPI you get the rank appended at the end, so instead of data.dmp, you end up with data_0.dmp and data_1.dmp.